### PR TITLE
BOM-1998 : Update Insights configuration to use Python 3.8

### PR DESCRIPTION
**Pull request by mzulqarnain1**
Relevant JIRA : https://openedx.atlassian.net/browse/BOM-1998

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

File: playbooks/roles/insights/defaults/main.yml [hunk 1]
```diff
@@ -226,8 +226,7 @@ insights_debian_pkgs:
   - libssl-dev # needed for mysqlclient python library
   - build-essential
   - gettext
-  - python3-pip
-  - python3-dev
+  - python3.8-dev
```
Comment (jmbowman) [inline playbooks/roles/insights/defaults/main.yml]: Won't you also need `python3.8-distutils`?  I thought virtualenv wasn't happy on Xenial without that.

Comment (mzulqarnain1) [reply playbooks/roles/insights/defaults/main.yml]: Added.

**Comment by jmbowman** [comment]
I don't see anything here or in the inherited `edx_service` role that adds the deadsnakes PPA so Python 3.8 can be installed.  Have you tried building this on a sandbox yet?

**Comment by mzulqarnain1** [comment]
> I don't see anything here or in the inherited `edx_service` role that adds the deadsnakes PPA so Python 3.8 can be installed. Have you tried building this on a sandbox yet?

@jmbowman Sandbox was successfully created here https://tools-edx-jenkins.edx.org/job/sandboxes/job/CreateSandbox/32616/

Logs from sandbox creation show that `python3.8-dev` and `python3.8-distutils` are indeed being installed from `deadsnakes ppa`, I think that is being added someplace before edx_service comes into play, maybe somewhere like https://github.com/edx/configuration/blob/master/playbooks/roles/jenkins_common/tasks/main.yml#L24 but someone from SRE team can tell us better.

**Comment by jmbowman** [comment]
Ah, it's working on the sandbox because it installs all the different services onto one box (and one of the other services adds the PPA before insights gets set up).  This will need to be fixed for insights to successfully install on a VM by itself.

**Comment by mzulqarnain1** [comment]
> Ah, it's working on the sandbox because it installs all the different services onto one box (and one of the other services adds the PPA before insights gets set up). This will need to be fixed for insights to successfully install on a VM by itself.

@jmbowman Opened https://github.com/edx/configuration/pull/5995 For this. After that PR is merged will make changes in this PR accordingly.

**Review by fredsmith — APPROVED** [review]

**Review by jmbowman — APPROVED** [review]

**Comment by natabene** [comment]
@mzulqarnain1 https://github.com/edx/configuration/pull/5995 is merged, please proceed.

**Comment by mzulqarnain1** [comment]
> @mzulqarnain1 #5995 is merged, please proceed.

@natabene I've already made the required changes. Just need owning team's review to merge. There is a JIRA open for that https://openedx.atlassian.net/browse/DENG-404

**Comment by natabene** [comment]
@mzulqarnain1 Got it. Hopefully next week team will be able to review.

**Review by macdiesel — APPROVED** [review]

**Pull request merged**